catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/LatLonCoordinateProcessorTest.java

```diff
@@ -0,0 +1,276 @@
+    assertThat(suggestions, empty());
+  }
+
+  @Test
+  public void testProcessorDetectsLatDecimalWithoutWhole() {
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), ".1234 34");
+    assertThat(cast(suggestions).get(0), hasCoordinates(from(0.1234, 34.0)));
+  }
+
+  @Test
+  public void testProcessorDetectsNegativeLatDecimalWithoutWhole() {
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), "-.1234 34");
+    assertThat(cast(suggestions).get(0), hasCoordinates(from(-0.1234, 34.0)));
+  }
+
+  @Test
+  public void testProcessorDetectsLatDecimal() {
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), "56.789 34");
+    assertThat(cast(suggestions).get(0), hasCoordinates(from(56.789, 34.0)));
+  }
+
+  @Test
+  public void testProcessorIgnoresTooLargeLatDecimal() {
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), "1234.1234 34");
+    assertThat(suggestions, empty());
+  }
+
+  @Test
+  public void testProcessorDetectsAndGracefullyHandlesMassiveDecimals() {
+    String massiveDecimal = "50.123456789012345678901234567890123456789012345678901234567890";
+    List<Suggestion> suggestions =
+        processor.enhanceResults(new LinkedList<>(), massiveDecimal + " " + massiveDecimal);
+    assertThat(suggestions, hasSize(1));
+  }
+
+  @Test
+  public void testRegexDetectsLatLonCommaSpaced() {
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), "12.153, -39.0352");
+    assertThat(cast(suggestions).get(0), hasCoordinates(from(12.153, -39.0352)));
+  }
+
+  @Test
+  public void testRegexDetectsLatLonParenthesis() {
+    List<Suggestion> suggestions =
+        processor.enhanceResults(new LinkedList<>(), "(12.153, -39.0352), (21.098, 19.148)");
+    assertThat(
+        cast(suggestions).get(0), hasCoordinates(from(12.153, -39.0352), from(21.098, 19.148)));
+  }
+
+  @Test
+  public void testRegexDetectsLatLonBrackets() {
+    List<Suggestion> suggestions =
+        processor.enhanceResults(new LinkedList<>(), "[12.153, -39.0352], [21.098, 19.148]");
+    assertThat(
+        cast(suggestions).get(0), hasCoordinates(from(12.153, -39.0352), from(21.098, 19.148)));
+  }
+
+  @Test
+  public void testRegexDetectsLatLonCurlyBraces() {
+    List<Suggestion> suggestions =
+        processor.enhanceResults(new LinkedList<>(), "{12.153, -39.0352}, {21.098, 19.148}");
+    assertThat(
+        cast(suggestions).get(0), hasCoordinates(from(12.153, -39.0352), from(21.098, 19.148)));
+  }
+
+  @Test
+  public void testRegexDetectsScatteredCoords() {
+    List<Suggestion> suggestions =
+        processor.enhanceResults(new LinkedList<>(), "blah;;;3g-12j.098     4.1");
+    assertThat(cast(suggestions).get(0), hasCoordinates(from(3.0, -12.0), from(0.098, 4.1)));
+  }
+
+  @Test
+  // TODO: Future improvement ... allow whitespace between relevant symbols: "- 14" & ".12"
```

Comment:
✏️ This is a valid potential improvement that can at least be discussed in the future. So include the ticket number with the TODO. Also note this might be fine as it stands. The ticket indicates the situation as such. 

https://codice.atlassian.net/browse/DDF-4242